svc/api/admin/src/route/cluster/datacenter.rs

```diff
@@ -0,0 +1,107 @@
+use api_helper::{anchor::WatchIndexQuery, ctx::Ctx};
+
+use proto::backend;
+use rivet_api::models;
+use rivet_convert::ApiInto;
+use rivet_operation::prelude::{
+	proto::{backend::pkg::cluster, chirp::response::Ok},
+	*,
+};
+use serde::Deserialize;
+use serde_json::{json, Value};
+
+use crate::auth::Auth;
+
+pub async fn create(
+	ctx: Ctx<Auth>,
+	cluster_id: Uuid,
+	body: models::AdminClusterDatacenterCreateRequest,
+) -> GlobalResult<models::AdminClusterDatacenterCreateResponse> {
+	let datacenter_id = Uuid::new_v4();
+	let name_id = Uuid::new_v4();
+
+	let dc = backend::cluster::Datacenter {
+		datacenter_id: Some(datacenter_id.into()),
+		cluster_id: Some(cluster_id.into()),
+		name_id: name_id.to_string(),
+		display_name: body.display_name.clone(),
+
+		provider: body.provider as i32,
+		provider_datacenter_id: body.provider_datacenter_id.clone(),
+		provider_api_token: None,
+
+		// When creating a datacenter, an empty pool of each type is added. This
+		// is to make sure that the datacenter starts in a valid state.
+		pools: vec![
+			backend::cluster::Pool {
+				pool_type: models::AdminPoolType::Job as i32,
```

Comment:
Should be using `backend::cluster::PoolType::XXX as i32`. Do the generated models automatically have a `ToPrimitive` implementation?